exammode/templates/exammode/base.html

```diff
@@ -0,0 +1,217 @@
+	<link rel="apple-touch-icon" sizes="72x72" href="{{ STATIC_URL }}favicons/apple-touch-icon-72x72.png">
+	<link rel="apple-touch-icon" sizes="144x144" href="{{ STATIC_URL }}favicons/apple-touch-icon-144x144.png">
+	<link rel="apple-touch-icon" sizes="60x60" href="{{ STATIC_URL }}favicons/apple-touch-icon-60x60.png">
+	<link rel="apple-touch-icon" sizes="120x120" href="{{ STATIC_URL }}favicons/apple-touch-icon-120x120.png">
+	<link rel="apple-touch-icon" sizes="76x76" href="{{ STATIC_URL }}favicons/apple-touch-icon-76x76.png">
+	<link rel="apple-touch-icon" sizes="152x152" href="{{ STATIC_URL }}favicons/apple-touch-icon-152x152.png">
+	<link rel="apple-touch-icon" sizes="180x180" href="{{ STATIC_URL }}favicons/apple-touch-icon-180x180.png">
+	<link rel="icon" type="image/png" href="{{ STATIC_URL }}favicons/favicon-192x192.png" sizes="192x192">
+	<link rel="icon" type="image/png" href="{{ STATIC_URL }}favicons/favicon-160x160.png" sizes="160x160">
+	<link rel="icon" type="image/png" href="{{ STATIC_URL }}favicons/favicon-96x96.png" sizes="96x96">
+	<link rel="icon" type="image/png" href="{{ STATIC_URL }}favicons/favicon-16x16.png" sizes="16x16">
+	<link rel="icon" type="image/png" href="{{ STATIC_URL }}favicons/favicon-32x32.png" sizes="32x32">
+	<meta name="msapplication-TileColor" content="#2b5797">
+	<meta name="msapplication-TileImage" content="{{ STATIC_URL }}mstile-144x144.png">
+</head>
+
+<body class="{% if profile %}{% if is_external_student %}external-student{% else %}internal-student{% endif %}{% endif %} lang-{{ request.LANGUAGE_CODE }}"
+ data-view-tag="{% block view_tag %}{% endblock %}">
+	<div class="page-wrap">
+		<nav class="topbar navbar navbar-inverse navbar-static-top">
+			{% if instance.language|first == "|" %}
+			<div class="container-fluid">
+				<ul class="user-menu nav navbar-nav navbar-right hidden-xs">
+					{% group_select %}
+					{% notification_menu %}
+					<li role="presentation" class="header">
+						<input style="margin: 15px" type="image" src="{% static 'images/uls.svg' %}" alt="{% trans 'Language' %}" class="btn btn-xs"
+						 id="lang-dropdown" type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" />
+						<ul class="dropdown-menu dropdown-menu-right" aria-labelledby="lang-dropdown">
+							{% for language in instance.language|list_unselected %}
+							<li>
+								<form action="{% url 'set_language' %}" method="post">
+									{% csrf_token %}
+									<input name="next" type="hidden" value="{{ request.get_full_path }}" />
+									<input name="language" type="hidden" value="{{ language }}" />
+									<button type="submit" class="btn">{{ language|language_name_local }}</button>
+								</form>
+							</li>
+							{% endfor %}
+						</ul>
+					</li>
+					<li>
+						<a style="display: inline-block; vertical-align: top;" class="btn btn-success" href="{% url 'exam_end' user.userprofile.active_exam.exam_taken.id %}">
```

Comment:
Finish exam buttom is now visible only in the multilingual courses ( `{% if instance.language|first == "|" %}` ).
Trying to add it outside of the if statement causes an error: `user.userprofile.active_exam.exam_taken.id` is not defined in the all templates that use the `base.html`. This can be fixed by checking the existence of the current course instance; it's absent when the exam is not yet selected. Once the exam is started, the finish exam buttom appears as it should.

Reply:
I've already fixed this issue.